### PR TITLE
Add Tabs engine to the sync manager

### PR DESCRIPTION
**Pull request by eoger**
Fixes https://github.com/mozilla-mobile/android-components/issues/5007.
This is fairly simple PR as it doesn't even need a direct dependency on the published a-s `tabs` component.

### CODE REVIEW

**Comment by grigoryk** [comment]
Please see https://github.com/mozilla-mobile/android-components/issues/4557 - we need to make sure that this will interact correctly with the `declined` list.

File: buildSrc/src/main/java/Dependencies.kt [hunk 1]
```diff
@@ -27,7 +27,7 @@ object Versions {
     const val disklrucache = "2.0.2"
     const val leakcanary = "1.6.3"
 
-    const val mozilla_appservices = "0.42.2"
+    const val mozilla_appservices = "0.44.0"
```
Comment (jonalmeida) [inline buildSrc/src/main/java/Dependencies.kt]: TODO me: run a smoke test against this upgrade to make sure native features are still working fine.

**Comment by eoger** [comment]
@grigoryk would the a-s sync manager take care of that?

File: buildSrc/src/main/java/Dependencies.kt (hunk 1 above)
Comment (jonalmeida) [reply buildSrc/src/main/java/Dependencies.kt]: Tested with the sample FxA app. Seems fine.

**Comment by jonalmeida** [comment]
@eoger just remember to put a PR up for Fenix to upgrade their a-s version when this lands.

**Comment by grigoryk** [comment]
> @grigoryk would the a-s sync manager take care of that?

I'm not really sure - need to think it through. Maybe? That's why there's an issue filed :-) If you can think through the implications here, it'd be great. I can take a close look at this early next week, as well.

**Comment by eoger** [comment]
> > @grigoryk would the a-s sync manager take care of that?
> 
> @eoger I'm not really sure - need to think it through. Maybe? That's why there's an issue filed :-) If you can think through the implications here, it'd be great. I can take a close look at this early next week, as well.

I did some digging around the code. This is what should happen when we add an engine:

- Before rust sync manager sync: `enabledChanges` [would not contain](https://searchfox.org/mozilla-mobile/rev/b56d507e9f715b69d94b324652e6b346de309812/android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt#365-379) our new engine, and [neither](https://searchfox.org/mozilla-mobile/rev/b56d507e9f715b69d94b324652e6b346de309812/android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt#381-384) `enginesToSync` (as it is an `intersection of stores for which we've set a 'handle' and those that are enabled.`)
- After the rust sync manager synced: If the user is connected on another device and disabled tabs, it would be in the resulting `declined` list, otherwise it would be in the `enabled` list [for the next sync](https://searchfox.org/mozilla-mobile/rev/b56d507e9f715b69d94b324652e6b346de309812/android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt#421-435) as `enabled = allEngines - declined`.

TL;DR: I believe we would respect the user's choice of declining the engine. Let me know if it makes sense!

**Comment by Amejia481** [comment]
bors retry

**Comment by grigoryk** [comment]
bors retry

**Comment by grigoryk** [comment]
Okay, since this bumps the A-S version, let's just wait until Monday; otherwise we'll break Fenix builds for the weekend.

**Comment by jonalmeida** [comment]
The last build seemed to have an intermittent test failure. Filed https://github.com/mozilla-mobile/android-components/issues/5213.
